Marlin/Marlin_main.cpp

```diff
@@ -1906,13 +1907,16 @@ static void clean_up_after_endstop_or_probe_move() {
 
     if (!endstops.z_probe_enabled) return;
 
+    // Make room for the probe
+    do_probe_raise(Z_RAISE_BETWEEN_PROBINGS);
```

Comment:
Aha, the code is trying to be too clever. Actually I think this should work in all cases:
```cpp
// Adjust the current Z and send it to the planner.
current_position[Z_AXIS] += z_tmp - real_z;
SYNC_PLAN_POSITION_KINEMATIC();
```